tests/webapps/pom.xml

```diff
@@ -0,0 +1,138 @@
+      </resource>
+    </resources>
+    <plugins>
+      <plugin>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <configuration>
+          <!-- 
+             Tests cannot and should not be validated with checkstyle.
+             these tests need to be flexible enough for good, bad, quirky
+             and everything in between with regards to compliance and
+             validity.
+             Having them pass checkstyle makes them invalid for integration
+             test cases
+             -->
+          <skip>true</skip>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>2.19.1</version>
+        <dependencies>
+          <dependency>
+            <groupId>org.apache.maven.surefire</groupId>
+            <artifactId>surefire-junit47</artifactId>
+            <version>2.19.1</version>
+          </dependency>
+        </dependencies>
+        <configuration>
+          <systemPropertyVariables>
+            <app.deploy.project>${app.deploy.project}</app.deploy.project>
+            <app.deploy.version>${maven.build.timestamp}</app.deploy.version>
+          </systemPropertyVariables>
+        </configuration>
+
+      </plugin>
+      <!-- This can be re-enabled when app-maven-plugin issue is fixed
+      https://github.com/GoogleCloudPlatform/app-maven-plugin/issues/85
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>0.1.2</version>
+        <configuration>
+          <dockerDirectory>${project.build.directory}/docker</dockerDirectory>
+        </configuration>
+        <executions>
+          <execution>
+            <id>deploy-integration-test</id>
+            <phase>integration-test</phase>
+            <goals>
+              <goal>deploy</goal>
+            </goals>
+            <configuration>
+              <deploy.project>${app.deploy.project}</deploy.project>
+              <deploy.version>${maven.build.timestamp}</deploy.version>
+              <deploy.promote>false</deploy.promote>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <version>1.8</version>
+        <executions>
+          <execution>
+            <id>cleanup-integration-test</id>
+            <phase>post-integration-test</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <target>
+                <exec executable="gcloud">
+                  <arg value="--quiet"/>
+                  <arg value="app"/>
+                  <arg value="versions"/>
+                  <arg value="delete"/>
+                  <arg value="${maven.build.timestamp}"/>
```

Comment:
You need to specify the service that the version belongs to here as well, unless you're using default service.